### DISCUSSION TITLE
`$request->resourceId` is null on action fields

### DISCUSSION THREAD

**Discussion by kichetof** [laravel/nova-issues#5164] [Q&A] [answered]
- Laravel Version: 9.43.0
- Nova Version: 4.19.4
- PHP Version: 8.1.13
- Database Driver & Version: Sail mysql
- Operating System and Version: Sail

### Description:
When you try to use `$request->resourceId` from an action fields, you get null when action is fired.
_(on debuging, I get the resourceId on index or detail loading, but null when action is launched)_

### Detailed steps to reproduce the issue on a fresh Nova installation:
`artisan nova:action CreateApiToken`

```php
public function fields(NovaRequest $request)
{
    ray($request->resourceId);
    ray()->showQueries();
    return [
        Text::make('Name')
            ->rules([
                'required',
                Rule::unique('personal_access_tokens', 'name')
                    ->where('tokenable_id', $request->resourceId),
            ])
    ];
}
```

My rule query was: 
```
select
  count(*) as aggregate
from
  `personal_access_tokens`
where
  `name` = 'test'
  and `tokenable_id` is null // should be an ID not null
````

**Comment by crynobone** [comment, marked as answer]
The best way to access Resource IDs is via `$request->resources` and the value will be based on context:

* Select all matching
  - `$request->resources` will return `"all"`
  - `$request->allResourcesSelected()` return `true`
  - `$request->selectedResourceIds()` return `null`
  - `$request->selectedResources()` return `null`
* Multiple resources selected:
  - `$request->resources` will return `[1, 2, 3, ...n]`
  - `$request->allResourcesSelected()` return `false`
  - `$request->selectedResourceIds()` return `[1, 2, 3, ..n]`
  - `$request->selectedResources()` return `Illuminate\Support\Collection<\Illuminate\Database\Eloquent\Model>`
* Single resource selected:
  - `$request->resources` will return `[1]`
  - `$request->allResourcesSelected()` return `false`
  - `$request->selectedResourceIds()` return `[1]`
  - `$request->selectedResources()` return `Illuminate\Support\Collection<\Illuminate\Database\Eloquent\Model>`

**Reply by kichetof** [reply]
Thanks for your great explanation! Really appreciate!
